File: src/day-18/part-2.ts
import fs from 'fs';
import { Input, parse } from './part-1';

function run() {
  const numbers = parse();

  let max = 0;

  for (let i = 0; i < numbers.length; i++) {
    for (let j = 0; j < numbers.length; j++) {
      if (i == j) continue;
      const magnitude = numbers[i].add(numbers[j]).reduce().magnitude();
      if (magnitude > max) max = magnitude;
    }
  }

  console.log(`Output: ${max}`);
}

if (!module.parent) {
  run();
}

File: rust/day-19/src/part2.rs
use itertools::Itertools;

use crate::part1::*;

pub fn largest_distance(input: &str) -> i32 {
    let scanners = parse(input);
    let solution = solve(&scanners);
    solution
        .iter()
        .cartesian_product(solution.iter())
        .map(|(a, b)| a.pos.subtract(&b.pos))
        .map(|p| p.magnitude())
        .max()
        .unwrap()
}
